exporters/include/opentelemetry/exporters/common/http/http_operation_curl.h

```diff
@@ -0,0 +1,534 @@
+      response_.memory = nullptr;
+      response_.size   = 0;
+    }
+    resp_headers_.clear();
+    resp_body_.clear();
+  }
+
+  /**
+   * Abort request in connecting or reading state.
+   */
+  void Abort()
+  {
+    is_aborted_ = true;
+    if (curl_ != nullptr)
+    {
+      // Simply close the socket - connection reset by peer
+      if (sockfd_)
+      {
+        ::close(sockfd_);
+        sockfd_ = 0;
+      }
+    }
+  }
+
+  CURL *GetHandle() { return curl_; }
+
+protected:
+  const bool raw_response_;         // Do not split response headers from response body
+  const size_t http_conn_timeout_;  // Timeout for connect.  Default: 5000ms
+
+  CURL *curl_;    // Local curl instance
+  CURLcode res_;  // Curl result OR HTTP status code if successful
+
+  http_sdk::EventHandler *callback_ = nullptr;
+
+  // Request values
+  http_sdk::Method method_;
+  std::string url_;
+  const Headers &request_headers_;
+  const http_sdk::Body &request_body_;
+  struct curl_slist *headers_chunk_ = nullptr;
+
+  // Processed response headers and body
+  std::vector<uint8_t> resp_headers_;
+  std::vector<uint8_t> resp_body_;
+
+  // Socket parameters
+  curl_socket_t sockfd_;
+
+  // long sockextr_   = 0;
+
+  curl_off_t nread_;
+  size_t sendlen_ = 0;  // # bytes sent by client
+  size_t acklen_  = 0;  // # bytes ack by server
+
+  std::future<long> result_;
+
+  /**
+   * Helper routine to wait for data on socket
+   *
+   * @param sockfd
+   * @param for_recv
+   * @param timeout_ms
+   * @return
+   */
+  static int WaitOnSocket(curl_socket_t sockfd, int for_recv, long timeout_ms)
+  {
+    struct timeval tv;
+    fd_set infd, outfd, errfd;
+    int res;
+
+    tv.tv_sec  = timeout_ms / 1000;
+    tv.tv_usec = (timeout_ms % 1000) * 1000;
+
+    FD_ZERO(&infd);
+    FD_ZERO(&outfd);
+    FD_ZERO(&errfd);
+
+    FD_SET(sockfd, &errfd); /* always check for error */
+
+    if (for_recv)
+    {
+      FD_SET(sockfd, &infd);
+    }
+    else
+    {
+      FD_SET(sockfd, &outfd);
+    }
+
+    /* select() returns the number of signalled sockets or -1 */
+    res = select((int)sockfd + 1, &infd, &outfd, &errfd, &tv);
+    return res;
+  }
+
+  // Raw response buffer
+  struct MemoryStruct
+  {
+    char *memory;
+    size_t size;
+  } response_;
+
+  /**
+   * Old-school memory allocator
+   *
+   * @param contents
+   * @param size
+   * @param nmemb
+   * @param userp
+   * @return
+   */
+  static size_t WriteMemoryCallback(void *contents, size_t size, size_t nmemb, void *userp)
+  {
+    size_t realsize          = size * nmemb;
+    struct MemoryStruct *mem = (struct MemoryStruct *)userp;
+
+    mem->memory = (char *)(realloc(mem->memory, mem->size + realsize + 1));
+    if (mem->memory == NULL)
+    {
+      /* out of memory! */
+      return 0;
+    }
+
+    memcpy(&(mem->memory[mem->size]), contents, realsize);
+    mem->size += realsize;
+    mem->memory[mem->size] = 0;
+
+    return realsize;
+  }
+
+  /**
+   * C++ STL std::string allocator
```

Comment:
That's a bit misleading. I don't see a `std::string` allocator here.

Reply:
Yes, I was trying to use std::string earlier for allocation, have remove the comment now.